demod_flex.c

```diff
@@ -623,16 +631,32 @@ static void parse_alphanumeric(struct Flex * flex, unsigned int * phaseptr, char
 
         int pt_offset;
 
-        if(flex_disable_timestamp)
-        {
-          pt_offset = sprintf(pt_out, "FLEX|%i/%i/%c/%c|%02i.%03i|%09lld",
-                        flex->Sync.baud, flex->Sync.levels, frag_flag, PhaseNo, flex->FIW.cycleno, flex->FIW.frameno, flex->Decode.capcode);
+        if (!json_mode) {
+          if(flex_disable_timestamp)
+          {
+            pt_offset = sprintf(pt_out, "FLEX|%i/%i/%c/%c|%02i.%03i|%09lld",
+                          flex->Sync.baud, flex->Sync.levels, frag_flag, PhaseNo, flex->FIW.cycleno, flex->FIW.frameno, flex->Decode.capcode);
+          }
+          else
+          {
+            pt_offset = sprintf(pt_out, "FLEX|%04i-%02i-%02i %02i:%02i:%02i|%i/%i/%c/%c|%02i.%03i|%09lld",
+                          gmt->tm_year+1900, gmt->tm_mon+1, gmt->tm_mday, gmt->tm_hour, gmt->tm_min, gmt->tm_sec,
+                          flex->Sync.baud, flex->Sync.levels, frag_flag, PhaseNo, flex->FIW.cycleno, flex->FIW.frameno, flex->Decode.capcode);
+          }
         }
-        else
-        {
-          pt_offset = sprintf(pt_out, "FLEX|%04i-%02i-%02i %02i:%02i:%02i|%i/%i/%c/%c|%02i.%03i|%09lld",
-                        gmt->tm_year+1900, gmt->tm_mon+1, gmt->tm_mday, gmt->tm_hour, gmt->tm_min, gmt->tm_sec,
-                        flex->Sync.baud, flex->Sync.levels, frag_flag, PhaseNo, flex->FIW.cycleno, flex->FIW.frameno, flex->Decode.capcode);
+        else {
+            // Lets just always send timestamp, the user can discard
```

Comment:
@lingfish @EliasOenal too bad I haven't seen this PR before. I thought I had all notifications turned on for this repo.

In the JSON it should have been the ISO format, just like the introduced by:

- https://github.com/EliasOenal/multimon-ng/pull/224

As that timestamp iso I think is not added now with this PR? 

There is no example output in the description to verify.

Reply:
Hrm, yeah, not sure as I live in a place where there's practically no FLEX to even test with.

Reply:
That PR added ISO timestamp format for all lines. It basically made the timestamp as part of the flex output irrelevant. 

It would have been great if --json would automatically use --iso1801 and not use the less accurate timestamp.

But that boat has passed as it's released but I'm not sure yet how it works as there is no example output.

I'll test next week what the current output will be and provide a PR to add iso timestamp format.